daft/execution/execution_step.py

```diff
@@ -5,6 +5,8 @@
 from dataclasses import dataclass, field
 from typing import Generic, TypeVar
 
+import fsspec
```

Comment:
This should be behind a `TYPE_CHECKING` guard since `fsspec` is an optional dependency.